### PR TITLE
Add icons only sidebar mode

### PR DESCRIPTION
**Pull request by Christian-Kr**
Hello to everyone,

I hope the work hasn't been done so far and I am not doing anything in parallel to someone else.

Nevertheless, I want the sidebar with the fancy background to have only icons and no text. So it gets a little bit smaller. I made the change here with the pull request. What is missing so far is the icon for smart playlists. There should be a different one to distinguish the smart and the custom playlists.

I hope that there is no big resistance for the change. I would really appreciate it. :-)

In case of any further questions, don't hesitate to contact me.

Thanks in advance.
Greetings Christian

### CODE REVIEW

File: src/widgets/fancytabbar.cpp [hunk 1]
```diff
@@ -156,7 +159,8 @@ void FancyTabBar::paintEvent(QPaintEvent *pe) {
 
   FancyTabWidget *tabWidget = qobject_cast<FancyTabWidget*>(parentWidget());
 
-  if (tabWidget->mode() != FancyTabWidget::Mode::LargeSidebar && tabWidget->mode() != FancyTabWidget::Mode::SmallSidebar) {
+  if (tabWidget->mode() != FancyTabWidget::Mode::LargeSidebar && tabWidget->mode() != FancyTabWidget::Mode::SmallSidebar &&
```
Comment (jonaski) [inline src/widgets/fancytabbar.cpp:162]: Same line or split all after `&&`

File: src/widgets/fancytabwidget.cpp [hunk 2]
```diff
@@ -225,7 +227,6 @@ void FancyTabWidget::SetMode(const Mode mode) {
 
   // There appears to be a bug in QTabBar which causes tabSizeHint to be ignored thus the need for this second shot repaint
   QTimer::singleShot(1ms, this, &FancyTabWidget::TabBarUpdateGeometry);
-
```
Comment (jonaski) [inline src/widgets/fancytabwidget.cpp:228]: Don't remove newlines

File: src/widgets/fancytabwidget.cpp [hunk 3]
```diff
@@ -355,7 +356,6 @@ void FancyTabWidget::paintEvent(QPaintEvent *pe) {
   }
 
   painter.drawPixmap(backgroundRect.topLeft(), pixmap);
-
```
Comment (jonaski) [inline src/widgets/fancytabwidget.cpp:358]: Don't remove newlines

File: src/widgets/fancytabwidget.cpp [hunk 4]
```diff
@@ -383,6 +383,7 @@ void FancyTabWidget::contextMenuEvent(QContextMenuEvent *e) {
     menu_ = new QMenu(this);
     QActionGroup *group = new QActionGroup(this);
     addMenuItem(group, tr("Large sidebar"), Mode::LargeSidebar);
+    addMenuItem(group, tr("Large sidebar without description"), Mode::IconOnlySideTabs);
```
Comment (jonaski) [inline src/widgets/fancytabwidget.cpp:386]: Can we call this something simpler, perhaps: `Icons sidebar` - `Mode::IconsSidebar` ?

**Review by jonaski — CHANGES_REQUESTED** [review]
Please squash corrections with the first commit.

Please update commit message, something like: `Add icons only sidebar mode`

File: src/widgets/fancytabbar.cpp (hunk 1 above)
Comment (Christian-Kr) [reply src/widgets/fancytabbar.cpp]: Done

**Review by jonaski — APPROVED** [review]

**Pull request merged**